### PR TITLE
Gutlunch - Simplemob Coolification

### PR DESCRIPTION
**Pull request by Nakalon**
## About The Pull Request

So I was playing and a chaplain produced a Gutlunch.
I tried feeding it raw meat because that's what was written in the description and couldn't.
Was very upset.

So I decided to make Gutlunches able to eat meat.
But then I realized, I could go much farther.
Much, much farther.

Made sprites for the baby and male gutlunch and dead sprites for all of them, so they wont just disappear on death anymore.
I gave them each their own type of udder, which means they create different chemicals.

Babies cant be milked, liquify what they eat and store the blood.
They will grow up only when their sac is full of blood.

Males produce mercury, acid and a tiny bit of blood.

Females produce the same milk they did before.

Gutlunches will now properly display the overlay for when they're full.

When killed they will pop like a balloon, releasing the contents of their udders and spawning gibs.

They can be butchered into crabmeat.

They will no longer continue to eat if their udder is full.

In the future I would like to have them lay eggs and other things but that's outside the scope of this PR.

## Why It's Good For The Game

Gutlunches were moderately forgotten.
This modernizes some of their code, fixes some things that were meant to work but didn't and reinforces sexual dimorphism.
Also hopefully gives the little bugs some use and makes it more interesting to be a sentient one, hopefully.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image

Here you can see a full male Gutlunch being popped open.
Do note I have since then updated the gibbing animation.

https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/88ebbe51-6279-497e-97e7-37b794220a57

Gutlunch genocide before I updated the gibbing.

https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/1483af11-3f09-4893-bc25-b52ce647965e

This is with the new gibbing animation.

https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/5e1f8bb1-16f8-4550-a197-9910d03548ec

</details>

## Changelog
:cl: PinkSuzuki
tweak: Tweaked Gutlunches, reinforced sexual dimorphism, allow them to eat meat and be butchered.
tweak: Male Gutlunches and Baby Gutlunches now have their own udders.
tweak: Gutlunches also now pop like a balloon when they die, releasing the content of their udders.
imageadd: Adds sprites for dead, Male and Baby Gutlunch.
fix: Gutlunch sacs now look bigger when they're full, as was intended.
code: Minorly tweaked some Gutlunch code
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
The secret ashlizard buff?? But yeah iirc the Gutmobs, at least the couple on the ashlizard nest only feed of gibs, organs and limbs left by the sacrifices, but this seems cool!

**Comment by Nakalon** [comment]
> The secret ashlizard buff?? But yeah iirc the Gutmobs, at least the couple on the ashlizard nest only feed of gibs, organs and limbs left by the sacrifices, but this seems cool!

yeah! It was meant to give it some flavor and make it so you can actually feed them with something more common!

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 1]
```diff
@@ -32,19 +32,23 @@
...
+	var/udder_type = /obj/item/udder/gutlunch
+	var/datum/component/udder/comp
+	var/obj/item/udder/gutlunch/udder
 
 /mob/living/simple_animal/hostile/asteroid/gutlunch/Initialize(mapload)
 	. = ..()
-	if(wanted_objects.len)
-		AddComponent(/datum/component/udder, /obj/item/udder/gutlunch, CALLBACK(src, PROC_REF(regenerate_icons)), CALLBACK(src, PROC_REF(regenerate_icons)))
+	AddComponent(/datum/component/udder, udder_type, CALLBACK(src, PROC_REF(regenerate_icons)), CALLBACK(src, PROC_REF(regenerate_icons)))
+	comp = GetComponent(/datum/component/udder)
```
Comment (itsmeow) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:50]: holding references to a component like this is a bad idea, we usually send data to and from components by signals. Basically, if you ever use GetComponent() you have made a mistake.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 2]
```diff
@@ -64,54 +68,71 @@
 		return TRUE
 
 	if(isobj(the_target) && is_type_in_typecache(the_target, wanted_objects))
-		return TRUE
+		if(udder.reagents.total_volume < udder.reagents.maximum_volume)
```
Comment (itsmeow) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:71]: you should use signals for something like this instead to replace the need for a reference.

You can create one specifically for the udder component:

COMSIG_MOB_IS_UDDER_FULL or something

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm [hunk 3]
```diff
@@ -64,54 +68,71 @@
...
 	if(isnull(gutlunch_full_overlay))
 		gutlunch_full_overlay = iconstate2appearance(icon, "gl_full")
-	if(new_udder_volume == max_udder_volume)
+	if(udder.reagents.total_volume == udder.reagents.maximum_volume && health > 0)
 		add_overlay(gutlunch_full_overlay)
 	..()
 
+/mob/living/simple_animal/hostile/asteroid/gutlunch/death()
+	. = ..()
+	chem_splash(get_turf(src), 1, list(udder.reagents))
```
Comment (itsmeow) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/gutlunch.dm:89]: instead, make a subtype of udder component that listens for death (via the signal) and spills the reagents. Or just add it to all udders

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by Nakalon** [comment]
I'm going to need help to adress these @itsmeow because I have no idea how to do that

**Comment by Nakalon** [comment]
converting this to draft while I don't fix what need fixing

**Comment by Rukofamicom** [comment]
This user has moved on from Bee and I don't believe this will be salvaged. Ping me if I'm mistaken.

**Pull request closed without merging**